src/components/Explore/TracesByService/Tabs/Structure/StructureScene.tsx

```diff
@@ -55,34 +60,88 @@ export class StructureTabScene extends SceneObjectBase<ServicesTabSceneState> {
       ...state,
     });
 
-    this.addActivationHandler(this._onActivate.bind(this));
+    this.addActivationHandler(() => {
+      this._onActivate();
+      return () => this._onDeactivate();
+    });
+  }
+
+  private _onDeactivate() {
+    if (this.updateTimeoutId) {
+      clearTimeout(this.updateTimeoutId);
+      this.updateTimeoutId = null;
+    }
+    this.lastProcessedData = null;
   }
 
   public _onActivate() {
     this.state.$data?.subscribeToState((state) => {
```

Comment:
BTW. is this subscription cleared? To get it automatically removed by scenes I think it should be wrapped in `this._subs.add(...)`

Reply:
Nice catch, added! 👍